### PR TITLE
Add `CONTRIBUTING.md` file

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add `Contribution.md` file

depends on https://github.com/eclipse-che/che-server/pull/559
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22243

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
Please, ignore OpenShift ci tests for this pr

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Contributing an SCM provider
```
Comment (ibuziuk) [inline CONTRIBUTING.md:1]: I think that `Contributing an SCM provider` should be put down and be the last section before CI

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# Sources build
+Run `mvn clean install` to build. Activate a faster profile build by adding `-Pfast`.
+
+# Image build and push
+1. Go to the `dockerfiles` directory.
+2. Run `./build.sh`.
+3. Tag the **che-server** image with your account: `docker tag quay.io/eclipse/che-server:next <docker registry>/<your account>/che-server:next`.
+4. Push the **che-server** image to your account: `docker push <docker registry>/<your account>/che-server:next`.
+
+# Start and debug
```
Comment (ibuziuk) [inline CONTRIBUTING.md:34]: Could we also add a `Developing within Eclipse Che` section ? similar to https://github.com/eclipse-che/che-dashboard/blob/main/CONTRIBUTING.md#developing-within-eclipse-che

File: CONTRIBUTING.md (hunk 1 above)
Comment (vinokurig) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (vinokurig) [reply CONTRIBUTING.md:36]: done

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@artaleks9 could you please review, since you did pair-programming / dogfooding recently during the Dev Tools week?

**Comment by artaleks9** [comment]
LGTM

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**